src/lightning_app/cli/lightning_cli.py

```diff
@@ -216,21 +217,24 @@ def cluster_logs(cluster_name: str, to_time: arrow.Arrow, from_time: arrow.Arrow
             f" Please select one of the following: [{', '.join(clusters.keys())}]"
         )
 
-    log_reader = _cluster_logs_reader(
-        client=client,
-        cluster_id=clusters[cluster_name],
-        start=from_time.int_timestamp,
-        end=to_time.int_timestamp,
-        limit=limit,
-        follow=follow,
-    )
+    try:
+        log_reader = _cluster_logs_reader(
+            logs_api_client=_ClusterLogsSocketAPI(client.api_client),
+            cluster_id=clusters[cluster_name],
+            start=from_time.int_timestamp,
+            end=to_time.int_timestamp,
+            limit=limit,
+            follow=follow,
+        )
 
-    colors = {"error": "red", "warn": "yellow", "info": "green"}
+        colors = {"error": "red", "warn": "yellow", "info": "green"}
 
-    for log_event in log_reader:
-        date = log_event.timestamp.strftime("%m/%d/%Y %H:%M:%S")
-        color = colors.get(log_event.labels.level, "green")
-        rich.print(f"[{color}]{log_event.labels.level:5}[/{color}] {date} {log_event.message.rstrip()}")
+        for log_event in log_reader:
+            date = log_event.timestamp.strftime("%m/%d/%Y %H:%M:%S")
+            color = colors.get(log_event.labels.level, "green")
+            rich.print(f"[{color}]{log_event.labels.level:5}[/{color}] {date} {log_event.message.rstrip()}")
```

Comment:
I see two potential things here:
- `try` block seems to be too large, it includes `logs_api_client` initialization and log reading loop; although I see we already have similar stories it the code (e.g. socket reading loop), maybe we can split it and make more specific exception processing blocks?
- we don't specify `Exception` type while we read logs in the loop. Does it hang while logs reading if a user uses `--follow` and tries to break the loop? Maybe we can specify potential exception types/ or at least process `KeyboardInterrupt` separately to correct break the loop?

Reply:
> we don't specify Exception type while we read logs in the loop. Does it hang while logs reading if a user uses --follow and tries to break the loop? Maybe we can specify potential exception types/ or at least process KeyboardInterrupt separately to correct break the loop

It used to hang before this PR, now this is fixed as well :0